src/types.rs

```diff
@@ -122,7 +124,7 @@ pub enum PredicateWitness {
 /// Prover's representation of the commitment secret: witness and blinding factor
 #[derive(Clone, Debug)]
 pub struct CommitmentWitness {
-    value: Scalar,
+    pub value: Scalar,
     blinding: Scalar,
```

Comment:
i guess both fields are safe to make public because there's no extra invariant to be enforced by the CommitmentWitness: the prover can choose both scalars arbitrarily.